src/cryptoadvance/specter/util/descriptor.py

```diff
@@ -415,6 +431,59 @@ def serialize(self):
 
         return AddChecksum(descriptor_open + base + descriptor_close)
 
+    def parse_signers(self, devices, cosigners_types):
+        from cryptoadvance.specter.key import Key
```

Comment:
Why not `from ..key import Key`, and also why here and not at the top of the file?

Reply:
@stepansnigirev Yeah, no reason for it not to be at the top of the file.

As for the dot import notation I find it great for sibling-level imports but I usually only use`..` when I'm working in something like Django where you're constantly drilling down into the identical directory structure copied over and over again for each child app. So you always know that, say, `..models` is still within that individual child app. 

I'm less of a fan when it reaches back to a more root level (e.g. `cryptoadvance.specter`). I'd rather be more explicit then. Rough analogy: If I'm inside an apartment, I don't have to specify my apartment number when I'm talking about my kitchen. But if I'm referring to something in the building's hallway or in someone else's apartment, then we should be explicit.

That being said, I can't really say I'm always super consistent one way or the other!

Given the above, since there was already another existing import with the full path notation, I left my import notation as-is but I did move it up.

Reply:
I understand your reasoning and agree with it.
Let's keep the import `from cryptoadvance.specter.key ...`, but move it up.

I like having all imports at the top of the file so it's easier to see the dependencies when moving / refactoring things, unless the dependency is used in a single function that is almost never called (for example during migration or in tests).